database/nostr-lmdb/src/store/mod.rs

```diff
@@ -86,64 +88,54 @@ impl Store {
     /// Is the event deleted
     pub fn event_is_deleted(&self, id: &EventId) -> Result<bool, Error> {
         let txn = self.db.read_txn()?;
-        let deleted: bool = self.db.is_deleted(&txn, id)?;
+        let deleted: bool = self.db.is_deleted(&txn, id.as_bytes())?;
         txn.commit()?;
         Ok(deleted)
     }
 
     pub fn count(&self, filter: Filter) -> Result<usize, Error> {
         let txn = self.db.read_txn()?;
-        let output = self.db.query(&txn, filter)?;
-        let len: usize = output.count();
+        let iter = self.db.query(&txn, filter)?;
+        let count = iter.count();
         txn.commit()?;
-        Ok(len)
+        Ok(count)
     }
 
     // Lookup ID: EVENT_ORD_IMPL
     pub fn query(&self, filter: Filter) -> Result<Events, Error> {
-        let mut events: Events = Events::new(&filter);
-
-        let txn: RoTxn = self.db.read_txn()?;
-        let output = self.db.query(&txn, filter)?;
-        events.extend(output.into_iter().map(|e| e.into_owned()));
+        let txn = self.db.read_txn()?;
+        let mut events_wrapper = Events::new(&filter);
+        events_wrapper.extend(self.db.query(&txn, filter)?.map(|e| e.into_owned()));
         txn.commit()?;
-
-        Ok(events)
+        Ok(events_wrapper)
     }
 
     pub fn negentropy_items(&self, filter: Filter) -> Result<Vec<(EventId, Timestamp)>, Error> {
         let txn = self.db.read_txn()?;
-        let events = self.db.query(&txn, filter)?;
-        let items = events
-            .into_iter()
-            .map(|e| (EventId::from_byte_array(*e.id), e.created_at))
+        let items = self
+            .db
+            .query(&txn, filter)?
+            .map(|e| (EventId::from_slice(e.id).unwrap(), e.created_at))
             .collect();
         txn.commit()?;
         Ok(items)
     }
 
     pub async fn delete(&self, filter: Filter) -> Result<(), Error> {
-        self.interact(move |db| {
-            let read_txn = db.read_txn()?;
-            let mut txn = db.write_txn()?;
-
-            db.delete(&read_txn, &mut txn, filter)?;
+        let (item, rx) = IngesterItem::delete_with_feedback(filter);
 
-            read_txn.commit()?;
-            txn.commit()?;
+        // Send to the ingester
+        self.ingester.send(item).map_err(|_| Error::MpscSend)?;
 
-            Ok(())
-        })
-        .await?
+        // Wait for a reply
+        rx.await?
     }
 
-    pub async fn wipe(&self) -> Result<(), Error> {
-        self.interact(move |db| {
-            let mut txn = db.write_txn()?;
-            db.wipe(&mut txn)?;
-            txn.commit()?;
-            Ok(())
-        })
-        .await?
+    pub fn wipe(&self) -> Result<(), Error> {
```

Comment:
The `write_txn` may block the current thread, for this reason was wrapped inside the `interact`